ext/AbstractDifferentiationForwardDiffExt.jl

```diff
@@ -61,20 +61,40 @@ function AD.hessian(ba::AD.ForwardDiffBackend, f, x::AbstractArray)
     return (ForwardDiff.hessian(f, x, cfg),)
 end
 
+function AD.value_and_derivative(::AD.ForwardDiffBackend, f, x::Real)
+    T = typeof(ForwardDiff.Tag(f, typeof(x)))
+    ydual = f(ForwardDiff.Dual{T}(x, one(x)))
+    return ForwardDiff.value(T, ydual), (ForwardDiff.extract_derivative(T, ydual),)
+end
+
 function AD.value_and_gradient(ba::AD.ForwardDiffBackend, f, x::AbstractArray)
     result = DiffResults.GradientResult(x)
     cfg = ForwardDiff.GradientConfig(f, x, chunk(ba, x))
     ForwardDiff.gradient!(result, f, x, cfg)
     return DiffResults.value(result), (DiffResults.derivative(result),)
 end
 
+function AD.value_and_second_derivative(ba::AD.ForwardDiffBackend, f, x::Real)
+    T = typeof(ForwardDiff.Tag(f, typeof(x)))
+    ydual, ddual = AD.value_and_derivative(ba, f, ForwardDiff.Dual{T}(x, one(x)))
+    return ForwardDiff.value(T, ydual), (ForwardDiff.extract_derivative(T, ddual[1]),)
+end
+
 function AD.value_and_hessian(ba::AD.ForwardDiffBackend, f, x)
     result = DiffResults.HessianResult(x)
     cfg = ForwardDiff.HessianConfig(f, result, x, chunk(ba, x))
     ForwardDiff.hessian!(result, f, x, cfg)
     return DiffResults.value(result), (DiffResults.hessian(result),)
 end
 
+function AD.value_and_derivatives(ba::AD.ForwardDiffBackend, f, x::Real)
+    T = typeof(ForwardDiff.Tag(f, typeof(x)))
+    ydual, ddual = AD.value_and_derivative(ba, f, ForwardDiff.Dual{T}(x, one(x)))
+    return ForwardDiff.value(T, ydual),
+    (ForwardDiff.value(T, ddual[1]),),
+    (ForwardDiff.extract_derivative(T, ddual[1]),)
```

Comment:
Similar to above, this will hit the fallback definition. I'd suggest
```suggestion
    xdual = ForwardDiff.Dual{T}(x, one(x))
    T2 = typeof(ForwardDiff.Tag(f, typeof(xdual)))
    ydual = f(ForwardDiff.Dual{T2}(xdual, one(xdual)))
    return ForwardDiff.value(T, ForwardDiff.value(T2, ydual)),
    (ForwardDiff.partials(T, ForwardDiff.value(T2, ydual), 1),),
    (ForwardDiff.partials(T, ForwardDiff.partials(T2, ydual, 1), 1),)
```
or equivalently
```suggestion
    xdual = ForwardDiff.Dual{T}(x, one(x))
    T2 = typeof(ForwardDiff.Tag(f, typeof(xdual)))
    ydual = f(ForwardDiff.Dual{T2}(xdual, one(xdual)))
    return ForwardDiff.value(T, ForwardDiff.value(T2, ydual)),
    (ForwardDiff.value(T, ForwardDiff.partials(T2, ydual, 1)),),
    (ForwardDiff.partials(T, ForwardDiff.partials(T2, ydual, 1), 1),)
```